controller/pod/podcontroller.go

```diff
@@ -0,0 +1,318 @@
+package pod
+
+/*
+Copyright 2017 - 2020 Crunchy Data Solutions, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	crv1 "github.com/crunchydata/postgres-operator/apis/cr/v1"
+	"github.com/crunchydata/postgres-operator/config"
+	"github.com/crunchydata/postgres-operator/kubeapi"
+	"github.com/crunchydata/postgres-operator/ns"
+	"github.com/crunchydata/postgres-operator/operator"
+
+	log "github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Controller holds the connections for the controller
+type Controller struct {
+	PodClient          *rest.RESTClient
+	PodClientset       *kubernetes.Clientset
+	PodConfig          *rest.Config
+	Ctx                context.Context
+	informerNsMutex    sync.Mutex
+	InformerNamespaces map[string]struct{}
+}
+
+// Run starts an pod resource controller
+func (c *Controller) Run() error {
+
+	err := c.watchPods(c.Ctx)
+	if err != nil {
+		log.Errorf("Failed to register watch for pod resource: %v", err)
+		return err
+	}
+
+	<-c.Ctx.Done()
+	return c.Ctx.Err()
+}
+
+// watchPods is the event loop for pod resources
+func (c *Controller) watchPods(ctx context.Context) error {
+	nsList := ns.GetNamespaces(c.PodClientset, operator.InstallationName)
+
+	for i := 0; i < len(nsList); i++ {
+		log.Infof("starting pod controller on ns [%s]", nsList[i])
+		c.SetupWatch(nsList[i])
+	}
+	return nil
+}
+
+// onAdd is called when a pod is added
+func (c *Controller) onAdd(obj interface{}) {
+
+	newPod := obj.(*apiv1.Pod)
+
+	newPodLabels := newPod.GetObjectMeta().GetLabels()
+	//only process pods with with vendor=crunchydata label
+	if newPodLabels[config.LABEL_VENDOR] == "crunchydata" {
+		log.Debugf("Pod Controller: onAdd processing the addition of pod %s in namespace %s",
+			newPod.Name, newPod.Namespace)
+	}
```

Comment:
Seems like this should be an early return as well.

```go
if newPodLabels[config.LABEL_VENDOR] != "crunchydata" {
  return
}

log.Debugf( ... )
```

Reply:
Updated to return early if the crunchy vendor label isn't found.